### PR TITLE
feat(index.js, __tests__): Support the Permission: directive

### PR DESCRIPTION
**Pull request by joker314**
The `Permission:` directive tells security researchers to avoid looking for security vulnerabilities. It must be provided no more than once, with a string value of `'none'`. This is tested for.

Some questions:
1. Is it a `feat` rather than a `fix`?

2. The validation code for the _Permission:_ directive is not split into a test for string-ness and a test for equality to `"none"`. Instead, both these tests are done in one go. This means a generic error message of `Expecting the string "none"` is shown, rather than `Expecting a string` or `Expecting "none"`. Given that only one value is allowed, is showing only one error acceptable?  (This is different to how the validation for _Encryption:_ is structured)

Also a note: the spec doesn't appear to clearly state whether `none` has to be in lowercase letters, but I assume that since URLs can be case-sensitive, all values must be.

Tested manually (and automatic tests added).

Also, I tried running `yarn run test:coverage` and it didn't work :( `error Command "test:coverage" not found.`

Thanks so much for all the support and help btw!

### CODE REVIEW

**Comment by lirantal** [comment]
1. Feat indeed since it's a new capability we're adding in.
2.1. Tests - looks good as is
2.2. 'none' value case sensitivity - I pinged Ed at https://twitter.com/liran_tal/status/1035175821990920195

The `test:coverage` is an old artifact that we can probably remove since jest is configured to automatically show coverage when you run the tests.  We should update the README to remove it.

**Review by lirantal — APPROVED** [review]
LGTM

**Comment by lirantal** [comment]
@joker314 see Ed's clarification in this PR referring to the case insensitive - https://github.com/securitytxt/security-txt/pull/127

**Comment by joker314** [comment]
Making it case-insensitive per the speedy response from Nightwatch Cyber

**Comment by lirantal** [comment]
Looks like all the other fields should also be case insensitive...

**Comment by joker314** [comment]
This is true! Looking through the code, ~~it looks like the only other place where this would affect us is the `Encryption:` directive's value. We check if it starts with `https`, but it's allowed to start with `HtTpS` and `HTTPS` too.~~
```javascript
if (options.encryption.toLowerCase().substr(0, 8) !== 'https://') {
```
We also should allow the directive to start with other stuff. The RFC gives these examples other than `https`:

> Example of a PGP key available from an OPENPGPKEY DNS:
> `Encryption: dns:5d2d37ab76d47d36._openpgpkey.example.com?type=OPENPGPKEY`
>
> Example of a PGP key being referenced by its fingerprint:
> `Encryption: openpgp4fpr:5f2de5521c63a801ab59ccb603d49de44b29100f`

Should this be addressed in this pull request or in another one?

**Comment by joker314** [comment]
Looks like the `Encryption:` directive is actually case-insensitive in our code already. I'll file an issue for the `Encryption:` directive being able to accept more values.

**Comment by lirantal** [comment]
Cool I think we can land this PR and do the schema change in another PR to not hold off this one.

**Comment by joker314** [comment]
So the Encryption: thing was accidentally fixed in 2.0.2. However, merging this PR will mean it will not have the Encryption directive fixed, but will have the Permission: directive fixed.

I suppose the commit should be put back into master, and if this triggers the bot to release a version which is identical (i.e. 2.0.2 and 2.0.3 will be the same) then I suppose it's not the worst thing in the world?

**Comment by joker314** [comment]
~~Oh, actually, I have an idea to undo this mess: I could add the fix which was mistakenly released in 2.0.2 into this PR, making merging it also put the commit back into the history?~~

EDIT: That's not guaranteed to work. I'll be squashing my commits down, so if I make multiple commits I'm not sure if this will trigger multiple releases?

So perhaps I should add the changes to this PR. The commit for 2.0.2 will never be in master's history again (but, it will still be hosted on GitHub), but the changes that it affected will be in 2.0.3. Should I do this?

**Comment by lirantal** [comment]
Closing and re-opening to trigger the CI again

**Comment by lirantal** [comment]
Let's land this PR as is, it will work ok.

**Comment by joker314** [comment]
I had to merge master into this branch or it wouldn't let me merge through the web interface. After tests complete, I'll squash and merge.

**Comment by lirantal** [comment]
Right that's how it should work now with PRs always being at the tip of the master branch so a FF merge is essential made.

**Pull request merged**
